### PR TITLE
[MNT] Added pytest flags to setup.cfg

### PR DESCRIPTION
**Pull request by aiwalter**
#### What does this implement/fix?
Consolidation of `pytest` flags

##### For all contributions
- [ ] I've added myself to the [list of contributors](https://github.com/alan-turing-institute/sktime/blob/main/.all-contributorsrc).
- [ ] Optionally, I've updated sktime's [CODEOWNERS](https://github.com/alan-turing-institute/sktime/blob/main/CODEOWNERS) to receive notifications about future changes to these files.
- [ ] I've added unit tests and made sure they pass locally.
- [ ] The PR title starts with either [ENH], [DOC] or [BUG] indicating whether the PR topic is related to enhancement, documentation or bug

##### For new estimators
- [ ] I've added the estimator to the online documentation.
- [ ] I've updated the existing example notebooks or provided a new one to showcase how my estimator works.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -28,7 +28,7 @@ test: ## Run unit tests
 	mkdir -p ${TEST_DIR}
 	cp .coveragerc ${TEST_DIR}
 	cp setup.cfg ${TEST_DIR}
-	cd ${TEST_DIR}; python -m pytest -v -n auto --showlocals --durations=20 $(PYTESTOPTIONS) --pyargs $(PACKAGE)
+	cd ${TEST_DIR}; python -m pytest
```
Comment (aiwalter) [inline Makefile:31]:
```suggestion
	cd ${TEST_DIR}; python -m pytest $ (PACKAGE)
```

**Comment by aiwalter** [comment]
..

**Comment by aiwalter** [comment]
@lmmentel @freddyaboulton could you two as our CI/CD experts pls have a look? 🙂

File: setup.cfg [hunk 2]
```diff
@@ -12,6 +12,15 @@ addopts =
     --ignore examples
     --ignore docs
     --doctest-modules
+    --durations 10
+    --timeout 300
```
Comment (aiwalter) [inline setup.cfg:16]:
```suggestion
    --timeout 600
```

**Comment by aiwalter** [comment]
for some reason when I edit the `build_wheels.sh` file to do the changes there as well, I get a `pre-commit` failure of the `shellcheck`. Should I just ignore that?

**Comment by lmmentel** [comment]
Great work, how about moving the configuration into `pyproject.toml`? `setup.py` is being flagged as legacy packaging backend so we might as well move the configuration.

File: setup.cfg [hunk 3]
```diff
@@ -12,10 +12,21 @@ addopts =
     --ignore examples
     --ignore docs
     --doctest-modules
+    --durations 10
+    --timeout 600
+    --cov sktime
+    --cov-report xml
+    --cov-report html
+    --showlocals
+    --pyargs
```
Comment (lmmentel) [inline setup.cfg:21]: I don't think this is needed since the invocation does not pass any arguments

File: setup.cfg [hunk 4]
```diff
@@ -12,10 +12,21 @@ addopts =
...
+    --verbose
+    -n 2
 filterwarnings =
     ignore::UserWarning
     ignore:numpy.dtype size changed
     ignore:numpy.ufunc size changed
+testpaths =
```
Comment (lmmentel) [inline setup.cfg:28]: this switches test discovery from package based to path based, have you verified that the same number of tests are discovered with and without this change?

File: setup.cfg (hunk 3 above)
Comment (aiwalter) [reply setup.cfg:21]: ok, I didnt know what `--pyargs` is used for, but let me then remove it

File: setup.cfg (hunk 4 above)
Comment (aiwalter) [reply setup.cfg:28]: yes I just verified that, its same

**Comment by aiwalter** [comment]
I think deprecating `setup.cfg` can be done in a separate PR.

**Review by fkiraly — APPROVED** [review]
Looks good to me - thanks for consolidating this!

@lmmentel, are you fine with this?

**Review by lmmentel — APPROVED** [review]

File: setup.cfg [hunk 5]
```diff
@@ -12,10 +12,21 @@ addopts =
     --ignore examples
     --ignore docs
     --doctest-modules
+    --durations 10
```
Comment (fkiraly) [inline setup.cfg:15]: the comment is no longer accurate, "ignore certain folders and pytest warnings". Perhaps one comment per block of things?

**Pull request merged**